-1 
Tested at: 279472d7a93de6de386cc29ff4a6d973d7626637
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 8 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>#define CommoTools_Utils_ExpressionSetter_h
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        CommonTools_Utils_ExpressionSetter_h
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_5_X_2015-09-09-2300/src/Fireworks/Core/src/FWHLTValidator.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_5_X_2015-09-09-2300/src/Fireworks/Core/src/FWISpyView.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_5_X_2015-09-09-2300/src/Fireworks/Core/src/FWGUIManager.cc:677:70: error: expression result unused [-Werror,-Wunused-value]
         parse("theta()", grammar.use_parser<1>() >> end_p, space_p).full;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_7_5_X_2015-09-09-2300/src/Fireworks/Core/src/FWGUIManager.cc:679:68: error: expression result unused [-Werror,-Wunused-value]
         parse("phi()", grammar.use_parser<1>() >> end_p, space_p).full;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11217/7992/summary.html

--- a/Fireworks/Core/src/FWGUIManager.cc
+++ b/Fireworks/Core/src/FWGUIManager.cc
@@ -665,30 +665,30 @@ FWGUIManager::showEDIFrame(int iToShow)
 void
 FWGUIManager::open3DRegion()
 {
-   FWModelId id =  *(m_context->selectionManager()->selected().begin());
-   float theta =0, phi = 0;
-   {
+   try {
+      FWModelId id = *(m_context->selectionManager()->selected().begin());
+      float theta =0, phi = 0;
       edm::TypeWithDict type = edm::TypeWithDict((TClass*)id.item()->modelType());
       using namespace boost::spirit::classic;
       reco::parser::ExpressionPtr tmpPtr;
       reco::parser::Grammar grammar(tmpPtr,type);
       edm::ObjectWithDict o(type, (void*)id.item()->modelData(id.index()));
-      try {
-         parse("theta()", grammar.use_parser<1>() >> end_p, space_p).full;
-         theta =  tmpPtr->value(o);
-         parse("phi()", grammar.use_parser<1>() >> end_p, space_p).full;
-         phi =  tmpPtr->value(o);
-
-         ViewMap_i it = createView( "3D Tower", m_viewSecPack->NewSlot());
-         FW3DViewBase* v = static_cast<FW3DViewBase*>(it->second);
-         v->setClip(theta, phi);
-         it->first->UndockWindow();
-         m_regionViews.push_back(v);
-      }
-      catch(const reco::parser::BaseException& e)
-      {
-         std::cout <<" FWModelFilter failed to base "<< e.what() << std::endl;
-      }
+      if (parse("theta()", grammar.use_parser<1>() >> end_p, space_p).full)
+         theta = tmpPtr->value(o);
+      else
+         throw FWExpressionException("syntax error", -1);
+      if (parse("phi()", grammar.use_parser<1>() >> end_p, space_p).full)
+         phi = tmpPtr->value(o);
+      else
+         throw FWExpressionException("syntax error", -1);
+      ViewMap_i it = createView( "3D Tower", m_viewSecPack->NewSlot());
+      FW3DViewBase* v = static_cast<FW3DViewBase*>(it->second);
+      v->setClip(theta, phi);
+      it->first->UndockWindow();
+   }
+   catch(const reco::parser::BaseException& e)
+   {
+      std::cout <<" FWModelFilter failed to base "<< e.what() << std::endl;
    }
 }
 

